governance/xc_admin/packages/xc_admin_common/src/programs/lazer/lazer_functions.ts

```diff
@@ -148,24 +154,143 @@ export function getDownloadableConfig(config: LazerConfig): DownloadableConfig {
  * @returns Object with validation result and optional error message
  */
 export function validateUploadedConfig(
-  existingConfig: DownloadableConfig,
-  uploadedConfig: unknown,
+  existingConfig: LazerState,
+  uploadedConfig: LazerState,
   cluster: PythCluster,
 ): ValidationResult {
-  // Basic validation logic for Lazer config
   try {
+    // Basic type validation
     if (typeof uploadedConfig !== "object" || uploadedConfig === null) {
       return {
         isValid: false,
         error: "Invalid JSON format for Lazer configuration",
       };
     }
 
-    // More detailed validation would be implemented here
-    // For now, return not implemented error
+    // Cast to LazerState
+    const uploadedLazerState = uploadedConfig as LazerState;
+
+    // Check for required fields
+    if (
+      !("shardId" in uploadedLazerState) ||
+      !("feeds" in uploadedLazerState) ||
+      !("publishers" in uploadedLazerState)
+    ) {
+      return {
+        isValid: false,
+        error: "Missing required fields in Lazer configuration",
+      };
```

Comment:
Is this not guaranteed by the type?

In general I recommend using something like [zod](https://zod.dev/) to describe and validate schemas in typescript; if this can't be guaranteed by the type system I'd use Zod to parse instead.